core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -808,9 +808,10 @@ void setListener(ServerStreamListener listener) {
     /**
      * Like {@link ServerCall#close(Status, Metadata)}, but thread-safe for internal use.
      */
-    private void internalClose(Throwable t) {
+    private void internalClose(Throwable t, String task) {
       // TODO(ejona86): this is not thread-safe :)
-      stream.close(Status.UNKNOWN.withCause(t), new Metadata());
+      String description = "Application Error @ task " + task;
```

Comment:
Considered ` @ context ` instead of ` @ task`. Let me know if you like one or another better.

Reply:
I would use `"Application Error while processing " + message/half close/onReady`.  The trace task name is too internally oriented for a good user message.

Reply:
@ejona86 what do you think?

Reply:
I'd suggest "Application error", "Application error processing RPC", or such. We have logged for the service authors to investigate what is wrong, we try not to tell the client too much because it could be an attacker, and the strings here are probably misleading. For that last point, the problem is the stubs morph things; most RPCs that throw will be within "half close" because they are unary and that's the point we call the application. But few know that. Including that information seems likely to just steer someone in the wrong direction.